### ISSUE TITLE
`chrome` and `browser` namespaces are not re-exported

### ISSUE THREAD

**Issue by BatMiles** [ghostery/adblocker#4120]
Using `@cliquz/adblocker-puppeteer` version `1.30.0` through npm

Node `v20.15.1`

the build command (basically equivalent to `tsc -p tsconfig.json`) errors out with the following:
```
node_modules/@cliqz/adblocker/dist/commonjs/request.d.ts:12:37 - error TS2503: Cannot find namespace 'browser'.

12 export type WebRequestTypeFirefox = browser.webRequest.ResourceType;
                                       ~~~~~~~

Found 1 error in node_modules/@cliqz/adblocker/dist/commonjs/request.d.ts:12

ERROR: "build:main" exited with 2.
```

tsconfig.json looks like:

```
{
  "compilerOptions": {
    "target": "es2017",
    "outDir": "build",
    "rootDir": "src",
    "moduleResolution": "node",
    "module": "commonjs",
    "declaration": true,
    "inlineSourceMap": true,
    "esModuleInterop": true /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */,
    /* Additional Checks */
    "noUnusedLocals": true /* Report errors on unused locals. */,
    "noUnusedParameters": true /* Report errors on unused parameters. */,
    "noImplicitReturns": true /* Report error when not all code paths in function return a value. */,
    "noFallthroughCasesInSwitch": true /* Report errors for fallthrough cases in switch statement. */,
    /* Debugging Options */
    "traceResolution": false /* Report module resolution log messages. */,
    "listEmittedFiles": false /* Print names of generated files part of the compilation. */,
    "listFiles": false /* Print names of files part of the compilation. */,
    "pretty": true /* Stylize errors and messages using color and context. */,
    "lib": ["es2017", "dom"],
    "types": ["jest", "node", "puppeteer", "winston", "chrome"],
    "typeRoots": ["node_modules/@types", "src/types"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["node_modules", "**/*.spec.ts", "__tests__"]
}
```

**Comment by BatMiles** [comment]
Solved this by installing the npm package `@types/firefox-webext-browser`, and including `firefox-webext-browser` in the `types` array in `tsconfig.json`

**Comment by seia-soto** [comment]
I'm reopening this issue to investigate internals and leave a note for developers or users of our library.

---

**This happens when you don't have `"skipLibCheck": true` in your `tsconfig.json`.**

TL;DR

This results are expected for TypeScript compilers as we don't ship some packages: `@types/chrome` and `@types/firefox-webext-browser` in our `dependencies` field (of `package.json`) even we're referencing types from those packages in our source code. Therefore, TypeScript compiler won't know what is `browser` and `chrome` global namespaces until we manually have those packages in our projects.

The details to the solution can be different depending on the project setup but I'd like to place those type only packages in `dependencies` field as we're re-exporting types.

refs https://stackoverflow.com/a/46011417

**Comment by seia-soto** [comment]
Update 2) From https://github.com/ghostery/adblocker/pull/4430, we're now embedding those types. So users will not be required to install types manually.

**Comment by seia-soto** [comment]
If any type conflict happens in the future, please open a new issue.
